core/src/main/scala/org/apache/spark/Accumulable.scala

```diff
@@ -49,6 +48,7 @@ import org.apache.spark.util.Utils
  * @tparam R the full accumulated data (result type)
  * @tparam T partial data that can be added in
  */
+@deprecated("use AccumulatorV2", "2.0.0")
```

Comment:
@rxin PS I think the `JavaSparkContext` needs similar treatment? methods like `intAccumulator` return the old API object.